solutions/src/main/java/ru/sberbank/school/task08/KryoManager.java

```diff
@@ -0,0 +1,74 @@
+package ru.sberbank.school.task08;
+
+import lombok.NonNull;
+import ru.sberbank.school.task08.state.GameObject;
+import ru.sberbank.school.task08.state.InstantiatableEntity;
+import ru.sberbank.school.task08.state.MapState;
+import ru.sberbank.school.task08.state.Savable;
+import ru.sberbank.school.util.Solution;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+@Solution(8)
+public class KryoManager extends SaveGameManager<MapState<GameObject>, GameObject> {
+    private Kryo kryo;
+
+    public KryoManager(@NonNull String filesDirectoryPath) {
+        super(filesDirectoryPath);
+    }
+
+    @Override
+    public void initialize() {
+        Kryo kryo = new Kryo();
+        KryoSerializer kryoSerializer = new KryoSerializer();
+        kryo.register(MapState.class, kryoSerializer);
+    }
+
+    @Override
+    public void saveGame(String filename, MapState<GameObject> gameState) throws SaveGameException {
+
+        try (FileOutputStream fos = new FileOutputStream(filename);
+             Output out = new Output(fos))
+        {
+            kryo.writeObject(out, gameState);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    @Override
+    public MapState<GameObject> loadGame(String filename) throws SaveGameException {
+        try (FileInputStream fis = new FileInputStream(filename);
+             Input in = new Input(fis)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-601877926>*